### PR TITLE
HW-54281 expose commit transaction into core

### PR DESCRIPTION
**Pull request by peter-joseph**

### CODE REVIEW

File: core/src/test/resources/transfer_commit_successful.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  "transition": "SCHEDULED",
+  "fromStatus": "QUOTED",
+  "toStatus": "SCHEDULED",
+  "notes": "Completing the Partial-Balance Transfer",
+  "links": [
+    {
+      "params": {
+        "rel": "self"
+      },
+      "href": "https://localhost-hyperwallet.aws.paylution.net:8181/rest/v3/transfers/trf-c6fa681a-b77e-4649-9c20-0dd789c8da80/status-transitions/sts-90e58807-c370-47d1-bf2c-bbfab380a1b9"
```
Comment (azakrevska-epam) [inline core/src/test/resources/transfer_commit_successful.json]: @peter-joseph may be better to use just "https://localhost:8181/rest/v3/transfers/trf-c6fa681a-b77e-4649-9c20-0dd789c8da80/status-transitions/sts-90e58807-c370-47d1-bf2c-bbfab380a1b9", what do you think?
(without -hyperwallet.aws.paylution.net)

File: core/src/test/resources/transfer_commit_successful.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+  "token": "sts-90e58807-c370-47d1-bf2c-bbfab380a1b9",
```
Comment (azakrevska-epam) [inline core/src/test/resources/transfer_commit_successful.json]: @peter-joseph May be better to use sts-123 token?

File: core/src/test/java/com/hyperwallet/android/transfer/CommitTransferTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        // verify
+        verify(mMockedStatusTransitionListener).onSuccess(mTransferCommitCaptor.capture());
+        verify(mMockedStatusTransitionListener, never()).onFailure(any(HyperwalletException.class));
+
+        // assert
+        RecordedRequest recordedRequest = mHwPlatform.getRequest();
+        String requestBody = recordedRequest.getBody().readUtf8();
+        assertThat(requestBody, is("{\"notes\":\"commit transfer test notes\",\"transition\":\"SCHEDULED\"}"));
+
```
Comment (azakrevska-epam) [inline core/src/test/java/com/hyperwallet/android/transfer/CommitTransferTest.java]: @peter-joseph Maybe we should check get/post for recordedRequest

File: core/src/main/java/com/hyperwallet/android/Hyperwallet.java [hunk 4]
```diff
@@ -764,6 +765,30 @@ public void listPrepaidCardReceipts(@NonNull final String prepaidCardToken,
...
     }
 
+    /**
+     * Commit the transfer recently created, identified by the transfer token, referenced from create transfer response
+     *
+     * @param transferToken transfer token generated when transfer is created
+     * @param notes         additional information for committing transfer
+     * @param listener      the callback handler of responses from the Hyperwallet platform; must not be null
+     */
+    public void commitTransfer(@NonNull final String transferToken, @Nullable final String notes,
```
Comment (fmattos-hw) [inline core/src/main/java/com/hyperwallet/android/Hyperwallet.java]: Please rename it to scheduleTransfer (don't forget the test class too). Thanks for adding the notes param, I missed that in the docs :) 🥇

**Pull request merged**